src/hooks/useFilterPossiblyMaliciousPositions.ts

```diff
@@ -0,0 +1,71 @@
+import { Token } from '@uniswap/sdk-core'
+import { useMemo } from 'react'
+import { PositionDetails } from 'types/position'
+import { hasURL } from 'utils/urlChecks'
+
+import { useDefaultActiveTokens } from './Tokens'
+import { useTokenContractsConstant } from './useTokenContractsConstant'
+
+function getUniqueAddressesFromPositions(positions: PositionDetails[]): string[] {
+  return Array.from(
+    new Set(positions.reduce<string[]>((acc, position) => acc.concat(position.token0, position.token1), []))
+  )
+}
+/*
+ * This function is an attempt to filter out an observed phishing attack from LP list UIs.
+ * Attackers would airdrop valueless LP positions with urls in the symbol to render phishing sites into users' LP position list view.
+ *
+ * Our approach to filtering these out without naively prohibiting all valid URL symbols is to:
+ * 1. allow any pair if both tokens are in the supported list
+ * 2. allow one url if one token is in the supported list
+ * 3. allow no urls if neither token is in the supported list
+ *
+ * The hope is that this approach removes the cheapest version of the attack without punishing non-malicious url symbols
+ */
+export function useFilterPossiblyMaliciousPositions(positions: PositionDetails[]): PositionDetails[] {
+  const activeTokensList = useDefaultActiveTokens()
+
+  const nonListPositionTokenAddresses = useMemo(
+    () => getUniqueAddressesFromPositions(positions).filter((address) => !activeTokensList[address]),
+    [positions, activeTokensList]
+  )
+
+  const symbolCallStates = useTokenContractsConstant(nonListPositionTokenAddresses, 'symbol')
+
+  const addressesToSymbol: Record<string, string> = useMemo(() => {
+    const result: Record<string, string> = {}
+    for (let i = 0; i < nonListPositionTokenAddresses.length; i++) {
+      const callResult = symbolCallStates[i].result
+      if (!callResult) continue
+      const address = nonListPositionTokenAddresses[i]
```

Comment:
complete side note - but i think having to correlate two separate arrays isn't necessarily ideal. this is a result of our contract call hooks so not worth doing anything here, but curious if we can make this a little nicer over time